core/src/accounts_hash_verifier.rs

```diff
@@ -55,23 +54,24 @@ impl AccountsHashVerifier {
                         break;
                     }
 
-                    match accounts_package_receiver.recv_timeout(Duration::from_secs(1)) {
-                        Ok(accounts_package) => {
-                            Self::process_accounts_package(
-                                accounts_package,
-                                &cluster_info,
-                                known_validators.as_ref(),
-                                halt_on_known_validators_accounts_hash_mismatch,
-                                pending_snapshot_package.as_ref(),
-                                &mut hashes,
-                                &exit,
-                                fault_injection_rate_slots,
-                                snapshot_config.as_ref(),
-                            );
-                        }
-                        Err(RecvTimeoutError::Disconnected) => break,
-                        Err(RecvTimeoutError::Timeout) => (),
+                    let accounts_package = pending_accounts_package.lock().unwrap().take();
+                    if accounts_package.is_none() {
+                        std::thread::sleep(Duration::from_millis(100));
```

Comment:
curious, any reason to decrease this to `100` ms compared to the original `1` second?

Reply:
Yeah, I was on the fence for this one. With the original channel, AHV isn't actively polling like in the new version. So the `accounts_package_receiver.recv_timeout(Duration::from_secs(1))` waits for up to one second before timing out, but then immediately loops back around and waits again. There was no sleep, but the channel I'm sure has some smarts/condvar that is handling this.

I'll switch this to sleep for one second.

Reply:
Done!

(Some more info: SnapshotPackagerService (SPS) also sleeps for 100 ms; I feel that, of the two, AHV should loop faster than SPS. Maybe AHV should be faster than 1 second?)

Reply:
We should only be making a snapshot every 100 roots right? So should be at least 40 seconds

Reply:
Yeah, the accounts hash interval is 100 slots by default, so yes, at least 40 seconds. Someone could set the accounts hash interval to 1 slot though (even though I doubt that'll happen). What about setting this to 400 ms via `SLOT_MS`?
https://github.com/solana-labs/solana/blob/302142bb257fbad64f13d7c72a6dca8cf91d36e3/sdk/program/src/clock.rs#L12-L13

Reply:
Sleep amount now set to `SLOT_MS`, aka 400 ms.